app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -179,6 +185,7 @@ class BrowserWebViewClient @Inject constructor(
                         }
                     }
                     if (isForMainFrame) {
+                        userAgentProvider.setHintHeader(webView.settings, url.toString())
```

Comment:
Should this be moved to inside `willOverrideUrl` where I believe we set UA also?

Reply:
We don’t set the UA there, that callback is only for Login detection logic.